pkgs/by-name/li/lib25519/package.nix

```diff
@@ -0,0 +1,80 @@
+{
+  stdenv,
+  lib,
+  python3,
+  fetchzip,
+  librandombytes,
+  libcpucycles,
+  system ? builtins.currentSystem,
+}:
+stdenv.mkDerivation (prev: {
+  pname = "lib25519";
+  version = "20240321";
+
+  src = fetchzip {
+    url = "https://lib25519.cr.yp.to/lib25519-${prev.version}.tar.gz";
+    hash = "sha256-R10Q803vCjIZCS4Z/uErsx547RaXfAELGQm9NuNhw+I=";
+  };
+
+  preConfigure = ''
+    patchShebangs configure
+    patchShebangs scripts-build
+  '';
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/lib25519-test
+  '';
+
+  # NOTE: lib25519 uses a custom Python `./configure`: it does not expect standard
+  # autoconfig --build --host etc. arguments: disable
+  configurePlatforms = [ ];
+  # Pass the hostPlatform string
+  configurePhase = ''
+    runHook preConfigure
+    ./configure --host=${system} --prefix=$out
+    runHook postConfigure
+  '';
```

Comment:
I guess you will either set `configurePlatforms = [ ]` or override `configurePhase`, but not both?